contact.sh

```diff
@@ -86,6 +86,35 @@ domain() {
     whois $OPTARG | grep "@$OPTARG" | tr -d "\t\r" | sort -u | sed 's/ //g'
     printf "\n"
 
+
+    # "EXPERIMENTAL" RECURSIVE WHOIS
+    # this tries to extract all emails from WHOIS ... note parsing is likely buggy so do not merge into master until further tested
+    printf "${GREEN}[+]${END} Checking WHOIS records recursively listing all addresses\n | Confidence level: ${RED}★${END} \n"
+
+    # FIXME: extracting the top domain needs more work ... there is probably already solutions for that
+    topdomain=$(echo $OPTARG | grep -o "[^\/\.]*\.\(com\|cn\|net\|co\.id\|vn\|se\|my\|cl\|eu\|com\.\(cn\|my\|tw\)\)$")
```

Comment:
Would something like this work?

```diff
+ topdomain=$(echo $OPTARG | grep -o '[^.]*\.[^.]*$')
- topdomain=$(echo $OPTARG | grep -o "[^\/\.]*\.\(com\|cn\|net\|co\.id\|vn\|se\|my\|cl\|eu\|com\.\(cn\|my\|tw\)\)$")
```

Reply:
Unfortunately, not. This would break for those double dot TLDs such as google.co.uk, etc.
The crappy parsing was done specifically for a list of sites we had, so it obviously is not complete.
What could be done would be, taking www.google.co.uk as example:
1. query for www.google.co.uk, which returns an error from WHOIS, however I don't know how to parse whether it is an error or not reliably from the WHOIS reply
2. query for google.co.uk, which works
3. query for co.uk which returns error (see 1.)
4. done

This would then work for all possible "double dot" TLDs even future ones without relying on a list, which will likely always stay incomplete due to new TLDs being added constantly.

... but I really thing (or at least hope) there is a better way, .e.g a lib that gets cuts of all subdomains from a given domain.
But in case the script must be called without any subdomains then setting `topdomain="$OPTARG"` would already work. But I have encountered stuff hosted only on a subdomain while the actual domain was completely unrelated to the subdomain. But such use cases may be out of scope for this script.